### ISSUE TITLE
[video_player_videohole] DASH - random jump to end of stream

### ISSUE THREAD

**Issue by mhoeckner** [flutter-tizen/plugins#573]
Hi all,

if we start a static playlist DASH/FMP4 stream, the player jumps random to the end of playlist and stops.
There is no specific time in stream when this occurs.

The stream itself works great in other players like Web Shaka 4.3.5 and ExoPlayer
But we all know that these players are all divas...

We tried several modifications in playlist but nothing helped. We extended logging - but no further informations. Any ideas why this is happening?

An example player I can provide here -> https://github.com/mhoeckner/flutter_tizen_test

Password I will send @haesik  and @swift-kim 

Thank you
regards
Markus

 flutter-tizen doctor -v
[!] Flutter (Channel unknown, 3.7.1, on macOS 13.3.1 22E261 darwin-x64, locale en-AT)
    ! Flutter version 3.7.1 on channel unknown at /Users/markushockner/development/flutter-tizen/flutter
      Currently on an unknown channel. Run `flutter channel` to switch to an official channel.
      If that doesn't fix the issue, reinstall Flutter by following instructions at https://flutter.dev/docs/get-started/install.
    ! Warning: `flutter` on your path resolves to /Users/markushockner/development/flutter/bin/flutter, which is not inside your current Flutter SDK checkout
      at /Users/markushockner/development/flutter-tizen/flutter. Consider adding /Users/markushockner/development/flutter-tizen/flutter/bin to the front of
      your path.
    ! Warning: `dart` on your path resolves to /Users/markushockner/development/flutter/bin/dart, which is not inside your current Flutter SDK checkout at
      /Users/markushockner/development/flutter-tizen/flutter. Consider adding /Users/markushockner/development/flutter-tizen/flutter/bin to the front of your
      path.
    ! Unknown upstream repository.
      Reinstall Flutter by following instructions at https://flutter.dev/docs/get-started/install.
    • Framework revision 7048ed95a5 (3 months ago), 2023-02-01 09:07:31 -0800
    • Engine revision 800594f1f4
    • Dart version 2.19.1
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Tizen toolchain - develop for Tizen devices
    • Framework revision e3fa9d40b4 (13 days ago), 2023-04-28 17:16:15 +0900
    • Engine revision 3a1a4b0df0
    • Tizen Studio 5.0 at /Users/markushockner/tizen-studio
    • .NET SDK 7.0.200 at /usr/local/share/dotnet/dotnet

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/markushockner/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/markushockner/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /usr/local/bin/chrome_unsafe

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.78.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.64.0

[✓] Connected device (3 available)
    • macOS (desktop)               • macos             • darwin-x64     • macOS 13.3.1 22E261 darwin-x64
    • Chrome (web)                  • chrome            • web-javascript • Google Chrome 113.0.5672.92
    • Tizen GU50AU8079UXZG (mobile) • 172.16.66.3:26101 • flutter-tester • Tizen 6.0

[✓] HTTP Host Availability
    • All required HTTP hosts are available

**Comment by hyue7** [comment]
> the player jumps random to the end of playlist and stops.
> There is no specific time in stream when this occurs.

Hi, @mhoeckner 
Could you please tell me how to reproduct this issue？

**Comment by mhoeckner** [comment]
Hi @hyue7 

thank you for fast response!

Within my example code - start stream 3

Our DASH streams needs auth - so you have to use my example code.

I do think you already got my email address - plz send me an email - I will send you the password. Or @swift-kim  and @haesik already got the needed password

**Comment by hyue7** [comment]
Hi， @mhoeckner 
I start play stream 3 till the end, but no error occurred. 
Could you describe this error? Or Maybe you can shoot a video and send to my email - yue7.huang@samsung.com,  so that i can accurately know what the error is.

**Comment by hyue7** [comment]
Hi, @mhoeckner 
I catch the error logs and see that:
DashCurl.cpp: CurlProcess(1657) > errorbuf : Operation timed out after 10002 milliseconds with 1424966 bytes received
DashCurl.cpp: ResultIfUnitCode(1730) >  HTTP TIME OUT ERROR-1 
DashCurl.cpp: ResultIf(1877) > [T01:55:36:859]  [Warning] CURLE_OPERATION_TIMEDOUT occured

This might be http request timeout ?

**Comment by mhoeckner** [comment]
Hi @hyue7 

thank you for the investigation. So the player jumps to end of playlist if a segment timeout occurs. Does the player any retries? Or is it a one time chance to fetch the segment?

Thank you
regards
Markus

**Comment by hyue7** [comment]
> Does the player any retries

Yes,  player have different retry times, the MaxRetryCount is 10:
`CDashDownLoadMgr.cpp: GetMaxRetryCount(2531) > [T01:55:36:860]  Debug:MaxRetryCount[10]`
but in this test app, player retry only once:
`CDashDownLoadMgr.cpp: retryDownloadSegment(2391) > retryDownloadSegment count[1]`
